github-actions/scan/src/configuration-builder.ts

```diff
@@ -51,56 +51,74 @@ export class SecHubConfigurationModelBuilderData {
-        model.codeScan = codescan;
+    if (isStringInArrayIgnoreCase(ScanType.CodeScan, builderData.scanTypes)) {
+        const codeScan: SecHubCodeScanConfiguration = {
+            use: [referenceName]
+        };
+        model.codeScan = codeScan;
     }
-    if (builderData.scanTypes?.indexOf(ScanType.LICENSE_SCAN) != -1) {
-        const licenseScan = new cm.LicenseScan();
-        licenseScan.use = [referenceName];
+    if (isStringInArrayIgnoreCase(ScanType.LicenseScan, builderData.scanTypes)) {
+        const licenseScan: SecHubLicenseScanConfiguration = {
+            use: [referenceName]
+        };
         model.licenseScan = licenseScan;
     }
-    if (builderData.scanTypes?.indexOf(ScanType.SECRET_SCAN) != -1) {
-        const secretScan = new cm.SecretScan();
-        secretScan.use = [referenceName];
+    if (isStringInArrayIgnoreCase(ScanType.SecretScan, builderData.scanTypes)) {
+        const secretScan: SecHubSecretScanConfiguration = {
+            use: [referenceName]
+        };
         model.secretScan = secretScan;
     }
-    if (builderData.scanTypes?.indexOf(ScanType.IAC_SCAN) != -1) {
-        const iacScan = new cm.IacScan();
-        iacScan.use = [referenceName];
+    if (isStringInArrayIgnoreCase(ScanType.IacScan, builderData.scanTypes)) {
+        const iacScan: SecHubIacScanConfiguration = {
+            use: [referenceName]
+        };
         model.iacScan = iacScan;
     }
 
     return model;
 }
 
-function createSourceOrBinaryDataReference(referenceName: string, builderData: SecHubConfigurationModelBuilderData, model: SecHubConfigurationModel) {
-    if (builderData.contentType == cm.ContentType.SOURCE) {
-
-        const sourceData1 = new cm.SourceData();
-        sourceData1.name = referenceName;
-
-        sourceData1.fileSystem.folders = builderData.includeFolders;
-        sourceData1.excludes = builderData.excludeFolders;
-
-        model.data.sources = [sourceData1];
-
-    } else if (builderData.contentType == cm.ContentType.BINARIES) {
-        const binaryData1 = new cm.BinaryData();
-        binaryData1.name = referenceName;
-
-        binaryData1.fileSystem.folders = builderData.includeFolders;
-        binaryData1.excludes = builderData.excludeFolders;
-
-        model.data.binaries = [binaryData1];
+function createSourceOrBinaryDataReference(referenceName: string, builderData: SecHubConfigurationModelBuilderData, model: SecHubConfiguration) {
+    if (!model.data) {
+        model.data = {};
     }
+    if (builderData.contentType === ContentType.SOURCE) {
+        const sourceData: SecHubSourceDataConfiguration = {
+            name: referenceName,
+            fileSystem: {
+                folders: builderData.includeFolders
+            },
+            excludes: builderData.excludeFolders
+        };
+        model.data.sources = [sourceData];
+    } else if (builderData.contentType === ContentType.BINARIES) {
+        const binaryData: SecHubSourceDataConfiguration = {
```

Comment:
This is wrong! There must be something like a generated  `SecHubBinaryDataConfiguration` which has to be used instead:
```suggestion
        const binaryData: SecHubBinaryDataConfiguration = {
```